### ISSUE TITLE
Ambigous input parsing (web3-eth-accounts)

### ISSUE THREAD

**Issue by holiman** [ethereum/web3.js#1342]
The API is a bit ambiguous. 

When strings are used for input instead if numbers, it's uncertain whether they get interpreted as hex-strings or decimal-strings. 

The following examples result in identical transactions: 
```javascript

account.signTransaction({'nonce':1, "value":0,"gas":10,"gasPrice":0, "chainId":1},"0x5384837ece911af85b8b2f2a0e6fe73f5f2add257fc25e1efbeebd4ee72226b7")
account.signTransaction({'nonce':1, "value":0,"gas":"10","gasPrice":0, "chainId":1},"0x5384837ece911af85b8b2f2a0e6fe73f5f2add257fc25e1efbeebd4ee72226b7")
account.signTransaction({'nonce':1, "value":0,"gas":"A","gasPrice":0, "chainId":1},"0x5384837ece911af85b8b2f2a0e6fe73f5f2add257fc25e1efbeebd4ee72226b7")
account.signTransaction({'nonce':1, "value":0,"gas":"0A","gasPrice":0, "chainId":1},"0x5384837ece911af85b8b2f2a0e6fe73f5f2add257fc25e1efbeebd4ee72226b7")
```

Consider the following example that a user may set up: 

```javascript
for (var nonce=9; nonce <= 16; nonce++){
  n = nonce.toString(16)
  console.log("Using nonce "+n)
  tx = account.signTransaction({'nonce':n, "value":0,"gas":10,"gasPrice":0, chainId:1},"0x5384837ece911af85b8b2f2a0e6fe73f5f2add257fc25e1efbeebd4ee72226b7");
  console.log(tx);
}
```
This will output the following transactions: 

```
[...]

Using nonce a
{ messageHash: '0xfda3444cd58e1e762082949a67bad4e45b6fde2d2ec9a8f23c8f6f5d87b87881',
  v: '0x26',
  r: '0xd83a784e94758c0b46f45dbbe3d7ed90f7a8ac66ce2bbcb3c0ae25e5a2e360bf',
  s: '0x790e1ace4be10c4d5690afcae25d4c2736a7f5d268034458fa51d1b5ce7d9f6d',
  rawTransaction: '0xf8490a800a80808026a0d83a784e94758c0b46f45dbbe3d7ed90f7a8ac66ce2bbcb3c0ae25e5a2e360bfa0790e1ace4be10c4d5690afcae25d4c2736a7f5d268034458fa51d1b5ce7d9f6d' }

[...]

Using nonce 10
{ messageHash: '0xfda3444cd58e1e762082949a67bad4e45b6fde2d2ec9a8f23c8f6f5d87b87881',
  v: '0x26',
  r: '0xd83a784e94758c0b46f45dbbe3d7ed90f7a8ac66ce2bbcb3c0ae25e5a2e360bf',
  s: '0x790e1ace4be10c4d5690afcae25d4c2736a7f5d268034458fa51d1b5ce7d9f6d',
  rawTransaction: '0xf8490a800a80808026a0d83a784e94758c0b46f45dbbe3d7ed90f7a8ac66ce2bbcb3c0ae25e5a2e360bfa0790e1ace4be10c4d5690afcae25d4c2736a7f5d268034458fa51d1b5ce7d9f6d' }

```
The "nonce a" and "nonce 10" (supposedly number 16 in hex-encoding) are identical.

**Comment by frozeman** [comment]
ok i will take a look, but for hex a 0x should be expected.

**Comment by holiman** [comment]
I can agree that whenever the string is prefixed with `0x` -- it's definitely hex. Since unprefixed strings are ambiguous, I'd advocate to _not_ accept those. 

Alternatively, always parse them as `dec`, and throw errors if it can't be interpreted in `dec` (because it contains a-f).

**Comment by frozeman** [comment]
will be fixed in the next release

**Comment by holiman** [comment]
Great! Mind posting a link to the fix / commit ?

**Comment by frozeman** [comment]
fixed in #097ceb8
